### PR TITLE
Allow fully overriding check and fmt commands

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

File: editors/code/package.json [hunk 1]
```diff
@@ -266,6 +273,13 @@
                     "default": "check",
                     "markdownDescription": "Cargo command to use for `cargo check`"
                 },
+                "rust-analyzer.checkOnSave.overrideCommand": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    },
+                    "markdownDescription": "Advanced option, fully override the command rust-analyzer uses for checking. The command must include `--message=format=json`."
```
Comment (Mark-Simulacrum) [inline editors/code/package.json:281]: Is this actually a requirement? Or do you just need JSON output?

Comment (matklad) [reply editors/code/package.json:281]: We need json output, which typically means `--message=format=json`. I guess I can phrase this better

**Pull request merged**